### DISCUSSION TITLE
Upgraded custom blocks / block palettes

### DISCUSSION THREAD

**Discussion by raltzy** [ScratchAddons/ScratchAddons#4292] [Ideas] [answered]
Ok so hear me out-
What if, when making a custom block, we had the option to make it appear in a certain palette (Thus, making the block the same color of that palette)
And after making the code for a block, we could have an option to hide the code that defines that block as well as show the code when right clicking on the block in the palette.
Speaking of right clicking the block in the palette, there should be an option for "locking" the block. This would make the block appear in every project you go to without having to import the code.
But we should also have the ability to show and hide blocks of any kind, whether we coded them or not. So when we right click on the palette, we have a list of all the blocks and we can click on any one of them to show or hide it.
This whole idea is a lot to take in and can be all over the place, so I made some images to recreate what I'm going for.
![Concept](https://user-images.githubusercontent.com/100053076/154824757-17342427-8761-4013-8f3b-23b756148131.png)
![Concept 1](https://user-images.githubusercontent.com/100053076/154824759-068ddeb2-5644-4879-8c5b-f8cb2b913d7b.png)
![Concept 2](https://user-images.githubusercontent.com/100053076/154824760-dfea0d8c-b4dd-43d2-bbab-b879ebf4aa70.png)
![Concept 3](https://user-images.githubusercontent.com/100053076/154824762-602c7265-01a2-4c55-aac0-3fffa2a59f28.png)

**Comment by Secret-chest** [comment]
Maybe add `<category>:` to the block name?
So `looks:previous costume would turn purple and show up at the bottom of the looks category

But this would be done automatically when you click a button

**Reply by kccuber-scratch** [reply]
what about non-SA users tho

**Comment by cobaltt7** [comment, marked as answer]
#1706  #4125
